src/app/(sidebar)/transaction/sign/components/SignWithWallet.tsx

```diff
@@ -36,52 +18,75 @@ export const SignWithWallet = ({
     updateSignedTx("");
     setSignError("");
     setSignSuccess("");
 
-    await kit.openModal({
-      onWalletSelected: async (option: ISupportedWallet) => {
-        try {
-          kit.setWallet(option.id);
-          const { address } = await kit.getAddress();
-          const networkType = getWalletNetwork(network.id);
-
-          const { signedTxXdr } = await kit.signTransaction(sign.importXdr, {
-            // You could send multiple public keys in case the wallet needs to handle multi signatures
-            address,
-            networkPassphrase: networkType,
+    if (walletKitPubKey) {
+      try {
+        const { signedTxXdr } =
+          await walletKitInstance.walletKit!.signTransaction(sign.importXdr, {
+            address: walletKitPubKey,
+            networkPassphrase,
           });
+        updateSignedTx(signedTxXdr);
+        setSignSuccess("1 signature(s) added");
+      } catch (error: any) {
+        if (error?.message) {
+          setSignError(error?.message);
+        }
+      }
+    } else {
+      // if a user didn't log in via stellar wallet kit in the main nav
+      // open a wallet kid modal to sign in
+      await walletKitInstance.walletKit!.openModal({
+        onWalletSelected: async (option: ISupportedWallet) => {
+          try {
+            walletKitInstance.walletKit?.setWallet(option.id);
+            const addressResult =
+              await walletKitInstance.walletKit?.getAddress();
+
+            if (addressResult?.address) {
+              updateWalletKitPubKey(addressResult.address);
 
-          updateSignedTx(signedTxXdr);
-          setSignSuccess("1 signature(s) added");
-        } catch (error: any) {
-          // the error for the following wallets:
-          // xbull
-          // albedo
-          // freighter
-          if (error?.message) {
-            setSignError(error?.message);
+              const result = await walletKitInstance.walletKit?.signTransaction(
+                sign.importXdr,
+                {
+                  // You could send multiple public keys in case the wallet needs to handle multi signatures
+                  address: addressResult.address,
+                  networkPassphrase,
+                },
+              );
+
+              if (result?.signedTxXdr) {
+                updateSignedTx(result.signedTxXdr);
+                setSignSuccess("1 signature(s) added");
+              } else {
+                throw { message: "couldn't sign with wallet. try again later" };
+              }
+            }
+          } catch (error: any) {
+            if (error?.message) {
+              setSignError(error?.message);
+            }
           }
-        }
-      },
-    });
+        },
+      });
+    }
   };
 
-  return (
+  return walletKitPubKey ? (
     <Button size="md" variant="tertiary" onClick={onSignWithWallet}>
-      Sign with wallet
+      {shortenStellarAddress(walletKitPubKey)}
```

Comment:
Should we use "Sign with AAAA...AAAA" instead? Just showing the address might not be clear what it's for.

![image](https://github.com/user-attachments/assets/8ecc7fec-5804-4aa4-86fa-39935555f9f1)

vs

![image](https://github.com/user-attachments/assets/477dabd7-5702-42f1-9eed-0ebb5ece3378)